gems/aws-sdk-core/spec/aws-sdk-core/credentials-provider-chain-tests.json

```diff
@@ -0,0 +1,599 @@
+        "boto2": "[Credentials]\naws_access_key_id = 0\naws_secret_access_key = 0",
+        "sdkStore": "\"[generated GUID]\" : {\n\t\"AWSAccessKey\" : \"0\"\n\t\"AWSSecretKey\" : \"0\"\n\t\"ProfileType\" : \"AWS\"\n\t\"DisplayName\" : \"default\""
+      },
+      "stsResponse": {
+        "accessKeyId": "0",
+        "secretAccessKey": "0",
+        "sessionToken": "0"
+      },
+      "ssoResponse": {
+        "accessKeyId": "0",
+        "secretAccessKey": "0",
+        "sessionToken": "0"
+      },
+      "processResponse": {
+        "accessKeyId": "0",
+        "secretAccessKey": "0",
+        "sessionToken": "0"
+      },
+      "containerResponse": {
+        "accessKeyId": "0",
+        "secretAccessKey": "0",
+        "sessionToken": "0"
+      },
+      "instanceResponse": {
+        "accessKeyId": "0",
+        "secretAccessKey": "0",
+        "sessionToken": "0"
+      }
+    },
+    "expected": {
+      "accessKeyId": "1",
+      "secretAccessKey": "2",
+      "sessionToken": "3"
+    }
+  },
+  {
+    "id": "3",
+    "documentation": "Environment variables: credentials",
+    "input": {
+      "env": {
+        "accessKeyId": "1",
+        "secretAccessKey": "2",
+        "sessionToken": "3",
+        "AWS_WEB_IDENTITY_TOKEN_FILE": "/some/path/",
+        "AWS_IAM_ROLE_ARN": "some-arn",
+        "AWS_CONTAINER_CREDENTIALS_FULL_URI": "https://awscredentials.amazonaws.com/credentials"
+      },
+      "configFiles": {
+        "aws": "[default]\naws_access_key_id = 0\naws_secret_access_key = 0\naws_session_token = 0\nsource_profile = A\nrole_arn = some-arn\nweb_identity_token_file = /some/path\nsso_session = dev\nsso_account_id = 12345678901\nsso_role_name = role\nsso_region = us-west-2\nsso_start_url = https://some.url\ncredential_process = some/process \n\n [profile A]\naws_access_key_id = 1\naws_secret_access_key = 2\naws_session_token = 3 \n\n [sso-session dev]\nsso_region = us-west-2\nsso_start_url = https://some.url",
+        "boto2": "[Credentials]\naws_access_key_id = 0\naws_secret_access_key = 0",
+        "sdkStore": "\"[generated GUID]\" : {\n\t\"AWSAccessKey\" : \"0\"\n\t\"AWSSecretKey\" : \"0\"\n\t\"ProfileType\" : \"AWS\"\n\t\"DisplayName\" : \"default\""
+      },
+      "stsResponse": {
+        "accessKeyId": "0",
+        "secretAccessKey": "0",
+        "sessionToken": "0"
+      },
+      "ssoResponse": {
+        "accessKeyId": "0",
+        "secretAccessKey": "0",
+        "sessionToken": "0"
+      },
+      "processResponse": {
+        "accessKeyId": "0",
+        "secretAccessKey": "0",
+        "sessionToken": "0"
+      },
+      "containerResponse": {
+        "accessKeyId": "0",
+        "secretAccessKey": "0",
+        "sessionToken": "0"
+      },
+      "instanceResponse": {
+        "accessKeyId": "0",
+        "secretAccessKey": "0",
+        "sessionToken": "0"
+      }
+    },
+    "expected": {
```

Comment:
Maybe not really related - but should account_id resolution be included in this as well?

Reply:
Possibly but I think we should test cred providers resolution only.